kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -895,7 +908,7 @@ private void encryptPlainSnapshots() throws Exception {
 
         for (Long snapshot : snapshots) {
             File fSnapshot = getSnapshotFile(snapshot);
-            if (!fSnapshot.exists()) {
+            if (fSnapshot == null || !fSnapshot.exists()) {
```

Comment:
this looks to me a genuine internal error (maybe KuraErrorCode.CONFIGURATION_ERROR). I would return KuraErrorCode.CONFIGURATION_SNAPSHOT_NOT_FOUND if I had specified a snapshot ID has a parameter to a method and the snapshot could not be find